From 734217599420766f9927f4d312577911dbf56791 Mon Sep 17 00:00:00 2001
From: "Anthony G. Basile" <blueness@gentoo.org>
Date: Sat, 8 Aug 2015 01:17:48 -0400
Subject: [PATCH 2/2] configure.ac: check for locale_t

fontforge/splinefont.h uses a "bad locale hack" for _WIN32 systems
where locale_t is not provided.  However, there are other situations
where locale_t may not be available.  For example, uClibc can be
configured to not provide locales.  A better approach is to check
for locale_t in configure.ac and enable the hack accordingly.

Signed-off-by: Anthony G. Basile <blueness@gentoo.org>
---
 configure.ac           | 1 +
 fontforge/splinefont.h | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ebeaf85..466214d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -387,6 +387,7 @@ gl_INIT
 AC_C_BIGENDIAN
 AC_TYPE_PID_T
 AX_PTHREAD
+AC_CHECK_TYPES([locale_t], [], [], [[#include <locale.h>]])
 
 #--------------------------------------------------------------------------
 # Check to see if "this" compiler can use these flags, errors, or warnings.
diff --git a/fontforge/splinefont.h b/fontforge/splinefont.h
index 973f343..4f1de12 100644
--- a/fontforge/splinefont.h
+++ b/fontforge/splinefont.h
@@ -3548,7 +3548,7 @@ char * delimit_null(const char * input, char delimiter);
 
 #include "ustring.h"
 
-#ifdef __MINGW32__
+#ifndef HAVE_LOCALE_T
 #define BAD_LOCALE_HACK
 typedef char* locale_t;
 #define LC_GLOBAL_LOCALE ((locale_t)-1)
-- 
2.4.6

